OpenRA.Mods.Common/Widgets/SpritePowerMeterWidget.cs

```diff
@@ -0,0 +1,65 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	public class SpritePowerMeterWidget : Widget
+	{
+		[Desc("The name of the Container Widget to tie the Y axis to.")]
+		[FieldLoader.Require]
+		public readonly string MeterAlongside = "";
+
+		[Desc("The name of the Container with the items to get the height from.")]
+		[FieldLoader.Require]
+		public readonly string ParentContainer = "";
+
+		[Desc("Height of each meter bar")]
+		[FieldLoader.Require]
+		public readonly int MeterHeight = 3;
+
+		[Desc("How many units of power each bar represents.")]
+		[FieldLoader.Require]
+		public readonly int PowerUnitsPerBar = 25;
+
+		[Desc("How many Ticks to wait before animating the bar.")]
+		[FieldLoader.Require]
+		public readonly int TickWait = 4;
+
+		[Desc("Blank Image for the meter bar")]
+		[FieldLoader.Require]
+		public readonly string NoPowerImage = "";
+
+		[Desc("When you have access power to use.")]
+		[FieldLoader.Require]
+		public readonly string AvailablePowerImage = "";
+
+		[Desc("Used power image")]
+		[FieldLoader.Require]
+		public readonly string UsedPowerImage = "";
+
+		[Desc("Too much power used meter image.")]
+		[FieldLoader.Require]
+		public readonly string OverUsedPowerImage = "";
+
+		[Desc("How many Ticks to wait before animating the bar.")]
+		[FieldLoader.Require]
+		public readonly string FlashPowerImage = "";
+
+		[Desc("The collection of images to get the meter images from.")]
+		[FieldLoader.Require]
+		public readonly string ImageCollection = "";
+
+		[ObjectCreator.UseCtor]
+		public SpritePowerMeterWidget() { }
```

Comment:
Having an empty widget as a placeholder for logic data is still a poor fit to our conventions - the logic classes generally take info via their own MiniYaml info dict.

The cleanest way to implement this will be to keep this widget and move the drawing logic back inside it using the `WidgetUtils` helpers to draw the sprites instead of using child widgets. See `ProductionTabsWidget` and `ProductionPaletteWidgets` for similar examples.

Reply:
It looks like those are also using widgets to draw the icons:

https://github.com/OpenRA/OpenRA/blob/23b3c237b7071fd308c4664b0b6c5d719c0f3c74/OpenRA.Mods.Common/Widgets/Logic/Ingame/ProductionTabsLogic.cs#L66-L82

Reply:
Not really - the `ProductionPaletteWidget` is drawing the icons, and this is adding the sidebar-specific frame over the top.